cocotb/clock.py

```diff
@@ -65,12 +65,43 @@ class Clock(BaseClock):
         signal: The clock pin/signal to be driven.
         period (int): The clock period. Must convert to an even number of
             timesteps.
-        units (str, optional): One of 
+        units (str, optional): One of
             ``None``, ``'fs'``, ``'ps'``, ``'ns'``, ``'us'``, ``'ms'``, ``'sec'``.
             When no *units* is given (``None``) the timestep is determined by
             the simulator.
+
+
+    If you need more features like a phase shift and an asymmetric duty cycle,
+    it is simple to create your own clock generator (that you then :func:`fork`):
+
+    .. code-block:: python
+
+        @cocotb.coroutine
+        def custom_clock():
+            high_time = Timer(high_delay)
+            low_time = Timer(low_delay)
+            yield Timer(initial_delay)
+            while True:
+                dut.clk <= 1
+                yield high_time
+                dut.clk <= 0
+                yield low_time
+
+    If you also want to change the timing during simulation,
+    use this slightly more inefficient example instead:
+
+    .. code-block:: python
+
+        @cocotb.coroutine
+        def custom_clock():
+            yield Timer(initial_delay)
+            while True:
+                dut.clk <= 1
+                yield Timer(high_delay)
+                dut.clk <= 0
+                yield Timer(low_delay)
```

Comment:
Not sure this is worth mentioning - without showing the other coroutine changing these parameters its quite hard to understand the difference.

Reply:
I have extended the comment above the code a bit to make it clearer.